### PR TITLE
Feat/cfe 481 view module mobile

### PR DESCRIPTION
**Pull request by jennieramida**
## Describe your changes

fixes: CFE-481, CFE-482

### CODE REVIEW

File: src/lib/pages/module-details/components/ModuleTop.tsx [hunk 1]
```diff
@@ -32,11 +32,10 @@ const baseTextStyle: TextProps = {
   whiteSpace: "nowrap",
 };
 
-export const ModuleTop = ({ moduleData, isVerified }: ModuleTopProps) => {
+const ModuleCta = ({ moduleData }: { moduleData: IndexedModule }) => {
```
Comment (Poafs1) [inline src/lib/pages/module-details/components/ModuleTop.tsx:35]: We can pass `moduleAddress` from `ModuleTop`, hence we no need to calculate it again.

File: src/lib/pages/interact/component/drawer/body/SelectModuleSection.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  filtered,
+  selectedModule,
+  setSelectedModule,
+  setStep,
+}: RenderModulesProps) => {
+  if (!modulesLength)
+    return <NoImageEmptyState desc="This address does not have any modules." />;
+
+  return filtered?.length ? (
+    <>
```
Comment (Poafs1) [inline src/lib/pages/interact/component/drawer/body/SelectModuleSection.tsx:40]: We don't need Fragment here.

File: src/lib/pages/interact/component/drawer/body/SelectFunctionSection.tsx [hunk 3]
```diff
@@ -88,15 +87,25 @@ export const ModuleFunctionBody = ({
...
+          {!isMobile && (
+            <Heading as="h6" variant="h6" fontWeight={600}>
+              {module.moduleName}
+            </Heading>
+          )}
+          {isMobile && (
+            <Heading as="h6" variant="h6" fontWeight={600} mt={6}>
+              Select View Function
+            </Heading>
+          )}
```
Comment (Poafs1) [inline src/lib/pages/interact/component/drawer/body/SelectFunctionSection.tsx:108]: Merge both to inline condition?

```sh
{isMobile ? () : ()}
```

File: src/lib/pages/interact/index.tsx [hunk 4]
```diff
@@ -18,27 +23,130 @@ import type { Addr, ExposedFunction, IndexedModule } from "lib/types";
...
+  );
+};
+
+export const ZeroState = ({
+  onOpen,
+  isMobile,
+}: {
+  onOpen: () => void;
+  isMobile: boolean;
+}) => {
```
Comment (Poafs1) [inline src/lib/pages/interact/index.tsx:108]: Please return it without keyword `return`

```js
=> (
  <Flex>
    ...
  </Flex>
)
```

File: src/lib/pages/interact/index.tsx [hunk 5]
```diff
@@ -18,27 +23,130 @@ import type { Addr, ExposedFunction, IndexedModule } from "lib/types";
...
+  setSelectedFn,
+  isZeroState,
+  handleDrawerOpen,
+}: {
+  selectedFn?: ExposedFunction;
+  setSelectedFn?: Dispatch<SetStateAction<ExposedFunction | undefined>>;
+  isZeroState?: boolean;
+  handleDrawerOpen?: (step: ModuleInteractionMobileStep) => void;
+}) => {
+  return (
```
Comment (Poafs1) [inline src/lib/pages/interact/index.tsx:47]: Same as below.

**Review by Poafs1 — COMMENTED** [review]
lgtm, can wait for others.

File: src/lib/pages/interact/component/drawer/body/SelectModuleSection.tsx (hunk 2 above)
Comment (jennieramida) [reply src/lib/pages/interact/component/drawer/body/SelectModuleSection.tsx:40]: I think we might need it

**Review by Poafs1 — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**